embedded-timestamp-mismatch/rewrite_embedded_bucket_id_control_min_mismatch.js

```diff
@@ -0,0 +1,213 @@
+  // temporary time-series collection in one batch and retry if any errors
+  // are encountered.
+  let retryTempInsert;
+  do {
+    retryTempInsert = false;
+    try {
+      tempTimeseriesBucketsColl.deleteMany({});
+      tempTimeseriesColl.insertMany(measurements);
+    } catch (e) {
+      print('An error occurred ' + e);
+      retryTempInsert = true;
+    }
+  } while (retryTempInsert);
+
+  // Run the bucket re-insertion in a transaction. It is necessary to
+  // interact with the buckets collection because transactions are not
+  // supported on the time-series view.
+  // Additionally, we want to retry this transaction on transient errors
+  // since we are touching potentially lots of data, which would cause
+  // excessive cache dirtying.
+  let hasTransientError;
+  do {
+    hasTransientError = false;
+    try {
+      const session = db.getMongo().startSession({retryWrites: true});
+      session.startTransaction();
+
+      const sessionBucketColl =
+          session.getDatabase(db.getName())
+              .getCollection('system.buckets.' + collName);
+      sessionBucketColl.deleteOne({_id: bucketId});
+      const bucketDocs = tempTimeseriesBucketsColl.find().toArray();
+      sessionBucketColl.insertMany(bucketDocs);
+
+      session.commitTransaction();
+    } catch (e) {
+      if (!shouldRetryTxnOnTransientError(e)) {
+        throw e;
+      }
+      hasTransientError = true;
+      print('Encountered a transient error. Retrying transaction.');
+      continue;
+    }
+  } while (hasTransientError);
+}
+
+function shouldRetryTxnOnTransientError(e) {
+  if ((e.hasOwnProperty('errorLabels') &&
+       e.errorLabels.includes('TransientTransactionError'))) {
+    return true;
+  }
+  return false;
+}
+
+//
+// Steps 1 & 2: Detect if a bucket has mismatched embedded bucket id timestamps
+// and control.min timestamps in the collection and re-inserts buckets with
+// these mismatches.
+//
+print(
+    'Re-inserting buckets that have a mismatched embedded bucket id timestamps and control.min timestamps in the collection ...\n');
+runFixEmbeddedBucketIdControlMinMismatchProcedure();
+tempTimeseriesBucketsColl.drop();
+
+//
+// Step 3: Validate that there are no buckets with mismatched embedded bucket id
+// timestamps and control.min timestamps in the collection.
+//
+print(
+    'Validating that there are no buckets that have a mismatched embedded bucket id timestamp and control.min timestamp ...\n');
+db.getMongo().setReadPref('secondaryPreferred');
+const validateRes = coll.validate({background: true});
+
+//
+// For v8.1+, buckets that have a mismatched embedded bucket id timestamp and
+// control.min timestamp will lead to a error during validation.
+//
+// Prior to v8.1, buckets that have a mismatched embedded bucket id timestamp
+// and control.min timestamp will lead to a warning during validation.
+//
+if (validateRes.errors.length != 0 || validateRes.warnings.length != 0) {
```

Comment:
Can we compare the actual error/warning output? I don't want unrelated warnings to report that there still exist buckets with mismatched _id

Reply:
Added checking for the correct log id for bucket version mismatch as well.